Dockerfile

```diff
@@ -7,29 +7,48 @@ RUN supercronicUrl=https://github.com/aptible/supercronic/releases/download/v0.1
     echo "$supercronicSha1sum  $supercronicBin" | sha1sum -c - && \
     chmod +x "$supercronicBin"
 
+ARG RAILS_ENV=production
+
 ENV PORT=3000 \
     SMTP_SERVER_PORT=2525 \
-    RAILS_ENV=production \
+    RAILS_ENV=${RAILS_ENV} \
     RAILS_LOG_TO_STDOUT=true \
     RAILS_SERVE_STATIC_FILES=true
 
 WORKDIR /usr/src/app
 
-COPY . ./
+RUN --mount=type=cache,target=/var/cache/apt/ \
+    buildDeps='libmagic-dev mariadb-server nodejs' && \
+    apt-get update && \
+    apt-get install --no-install-recommends -y $buildDeps
```

Comment:
This is happening in the base stage, so e.g. mariadb-server and the other `buildDeps` are getting into the final image.